PWGHF/Tasks/HFMCValidation.cxx

```diff
@@ -278,15 +311,27 @@ struct ValidationRecLevel {
   using HfCandProng2WithMCRec = soa::Join<aod::HfCandProng2, aod::HfCandProng2MCRec>;
   using HfCandProng3WithMCRec = soa::Join<aod::HfCandProng3, aod::HfCandProng3MCRec>;
 
-  void process(HfCandProng2WithMCRec const& cand2Prongs, HfCandProng3WithMCRec const& cand3Prongs, aod::BigTracksMC const& tracks, aod::McParticles const& particlesMC)
+  void process(HfCandProng2WithMCRec const& cand2Prongs, HfCandProng3WithMCRec const& cand3Prongs, aod::BigTracksMC const& tracks, aod::McParticles const& particlesMC, aod::McCollisions const& mcCollisions, soa::Join<aod::Collisions, aod::McCollisionLabels> const& collisions)
   {
     // loop over tracks
     for (auto& track : tracks) {
       uint index = uint(track.collisionId() >= 0);
       if (track.has_mcParticle()) {
         auto particle = track.mcParticle(); // get corresponding MC particle to check origin
+        uint index2 = 0;
+        if (index) {
+          auto collision = collisions.rawIteratorAt(track.collisionId());
+          auto mcCollision = mcCollisions.rawIteratorAt(particle.mcCollisionId());
```

Comment:
Can you not get the collisions directly via the getters `track.collision()`, `particle.mcCollision()` (or `_as<>()`)?

Reply:
I did it for the `mcCollision` but if I do it also for the collision for some reason then I cannot call `mcCollisionId()` even if I joined the collision table with the one with MC collision labels (so I can only address your comment partially :-))

Reply:
Are you saying that if you do `auto collision = track.collision_as<CollisionsWithMCLabels>();`, calling `collision.mcCollisionId()` then crashes?

Reply:
With `collision_as` indeed works